packages/transaction-controller/CHANGELOG.md

```diff
@@ -5,8 +5,55 @@ The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
+- Add `onBootCleanup` functionality to constructor and `onNetworkChange` ([#1916](https://github.com/MetaMask/core/pull/1916))
+  - Initialises approvals of `TransactionStatus.unapproved` transactions on active network
+  - Refreshes the gas values for `TransactionStatus.unapproved` transactions on active network
+  - Submits the `TransactionStatus.approved` transactions on active network
+- Support advanced gas fees in `TransactionController` ([#1966](https://github.com/MetaMask/core/pull/1966))
+  - `getSavedGasFees` callback added as an optional parameter in constructor.
+  - Setting `getSavedGasFees` callback will make saved gas fees (`advancedGasFees` in the `PreferencesController` state) be taken into account when updating gas fees.
+- Add updateCustodialTransaction to update the custodial transactions associated with a specific transaction ID. ([#2018](https://github.com/MetaMask/core/pull/2018))
+- Add `accessList` to txParam types ([#2016](https://github.com/MetaMask/core/pull/2016))
+- Adds `estimateGasBuffered` public method ([#2021](https://github.com/MetaMask/core/pull/2021))
+- Add method for adding params to transaction meta ([#1985](https://github.com/MetaMask/core/pull/1985))
+- Add gas values validation to transaction params ([#1978](https://github.com/MetaMask/core/pull/1978))
+- Add `approveTransactionsWithSameNonce` public method to core transaction controller ([#1961](https://github.com/MetaMask/core/pull/1961))
+- Add `clearUnapprovedTxs` public method to `TransactionController` ([#1979](https://github.com/MetaMask/core/pull/1979))
+- Add `updatePreviousGasParams` public method to `TransactionController` ([#1943](https://github.com/MetaMask/core/pull/1943))
+- Add `TransactionEvents` to be emitted from `TransactionController` ([#1894](https://github.com/MetaMask/core/pull/1894))
+- Populate the `firstRetryBlockNumber`, `retryCount`, and `warning` properties in the transaction metadata. ([#1896](https://github.com/MetaMask/core/pull/1896))
+
 ### Changed
+- **BREAKING:** Pending transactions are now automatically resubmitted. ([#1896](https://github.com/MetaMask/core/pull/1896))
+  - This can be disabled by setting the new `pendingTransactions.isResubmitEnabled` constructor option to `false`.
+- **BREAKING:** Bump dependency and peer dependency on `@metamask/network-controller` to ^16.0.0
+- Persist specific error properties in core transaction metadata ([#1915](https://github.com/MetaMask/core/pull/1915))
+  - create `TxError` type with specific properties to support both clients.
+  - Updated `failTransaction` to use the new type.
+- Align core transaction error messages with extension ([#1980](https://github.com/MetaMask/core/pull/1980))
+  - `processApproval` when the user rejects the request: MetaMask Tx Signature: User denied transaction signature.`
```

Comment:
I'd vote to remove the nested list of individual message updates as it feels excessive.